toolbox/hgfs/archive.go

```diff
@@ -229,42 +229,36 @@ func archiveRead(u *url.URL, tr *tar.Reader) error {
 
 // archiveWrite writes the contents of the given source directory to the given tar.Writer.
 func archiveWrite(u *url.URL, tw *tar.Writer) error {
-	dir := filepath.Dir(u.Path)
 
 	return filepath.Walk(u.Path, func(file string, fi os.FileInfo, err error) error {
 		if err != nil {
 			return filepath.SkipDir
 		}
 
-		name := strings.TrimPrefix(file, dir)[1:]
-
-		header, _ := tar.FileInfoHeader(fi, name)
-
-		header.Name = name
-
-		if header.Typeflag == tar.TypeDir {
-			header.Name += "/"
+		header, err := tar.FileInfoHeader(fi, fi.Name())
+		if err != nil {
+			return err
 		}
+		header.Name = strings.TrimPrefix(
+			strings.Replace(file, filepath.Dir(u.Path), "", -1),
+			string(filepath.Separator))
 
-		var f *os.File
+		if err := tw.WriteHeader(header); err != nil {
+			return err
+		}
 
 		if header.Typeflag == tar.TypeReg && fi.Size() != 0 {
-			f, err = os.Open(file)
+			f, err := os.Open(file)
```

Comment:
Now the file isn't closed at all, so this leaks file descriptors.

The ignored errors from FileInfoHeader and WriteHeader were intentional, the additional checks aren't needed afaict.

And didn't see a response to this question in the first review "Can't tell how this is different without trying it, or what exactly was broken. Can you give some examples?"